pkg/services/pluginsintegration/plugininstaller/service.go

```diff
@@ -18,26 +19,36 @@ type Service struct {
 	log             log.Logger
 	pluginInstaller plugins.Installer
 	pluginStore     pluginstore.Store
+	async           bool
 }
 
-func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles, pluginStore pluginstore.Store, pluginInstaller plugins.Installer) *Service {
+func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles, pluginStore pluginstore.Store, pluginInstaller plugins.Installer) (*Service, error) {
 	s := &Service{
 		features:        features,
 		log:             log.New("plugin.backgroundinstaller"),
 		cfg:             cfg,
 		pluginInstaller: pluginInstaller,
 		pluginStore:     pluginStore,
+		async:           cfg.InstallPluginsAsync,
 	}
-	return s
+	if !cfg.InstallPluginsAsync {
+		// Block initialization process until plugins are installed
+		err := s.installPlugins(context.Background())
```

Comment:
Maybe we should add a timeout to this `context` so in case of a timeout it returns a "context deadline exceeded" error, rather than hanging forever?

Reply:
Yeah this is all legacy code that was copied from the CLI and moved here IIRC - certainly not a function to get inspiration from considering the weird panic/recover stuff 😅 I would say there is probably no harm having a context with timeout here and maybe we can create a follow-up issue to tidy up the current setup inside the repo client.

Reply:
Added a timeout of 5 minutes for this which should be more than enough!